### DISCUSSION TITLE
Remove force-push restriction from contribution guidelines

### DISCUSSION THREAD

**Discussion by krichter722** [renovatebot/renovate#30229] [Suggest an Idea]
### Tell us more.

I've not been in touch with renovate development so far, however I notice that the section https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#do-not-force-push-to-your-pull-request-branch does not make a lot of sense after GitHub feature [mark as viewed](https://github.blog/2019-07-01-mark-files-as-viewed/) has been introduced. The amount of files to review should be the same, no matter whether you push or force-push, because the chances are detected by GitHub. I think, the section can be removed without replacement.

**Comment by rarkins** [comment]
Force pushing makes it harder to understand the history too, so I would still like people to avoid it. In which scenarios do you consider it beneficial?

**Reply by HonkingGoose** [reply]
I don't like force pushes, because they:

- Make it hard to confirm that my docs suggestions are in the new force-pushed commit
- Strip the "commit co-authored by HonkingGoose" credit from my review suggestions
- Make it hard to understand the history of the work, and the thinking behind each commit

Our rules allow force pushes in limited cases:

> Force pushing a PR, or using the "Update with rebase" button is OK when you:
> 
> - make large changes on a PR which require a full review anyway
> - bring the branch up-to-date with the target branch and incorporating the changes is more work than to create a new PR

**Comment by krichter722** [comment]
I guess it's beneficial only locally where it's easier to rebase a single commit rather than a commit range, especially in a chain of branches. Good points about unchanged commits in the PR history. I guess, everything is fine as it is :)
